### DISCUSSION TITLE
Touch and external displays not working on T440

### DISCUSSION THREAD

**Discussion by thomas-catt** [MultimediaLucario/Lenovo-ThinkPad-T440s#29] [Q&A]
Used this Monterey EFI on my T440 for a few days, but the touch screen wasn’t working, and so wasn’t external display support. My T440 has a broken touchpad and spotty keyboard, so I don’t use them, but I remember my keyboard wasn’t functioning either (even the working keys). Rest of the install was smooth and stable.

**Comment by MultimediaLucario** [comment]
If you are looking to patch a touch display, I highly recommend you try reading the Post-Install Tweaks section of the guide on how to do so.

**Reply by MultimediaLucario** [reply]
It's mainly the reason why I just use the non touch 1600x900 display as it requires no additional patches.

**Reply by thomas-catt** [reply]
I’m using a stock display, never got it replaced or anything. Isn’t patching only required when you replace your display with an aftermarket one? (to fix the garbled screen)

Also, any thoughts on external display not working? I suspect all of this could be related to USB mapping, since once the hackintosh unexpectedly restarted in it’s sleep too (and I was getting some mouse lag issues related to mapping).

**Reply by MultimediaLucario** [reply]
Variants with a touchscreen do require addtional patching. Please read the Post-Install Tweaks section of my guide for more information.

**Reply by thomas-catt** [reply]
I see. Thank you for your reply. Also, do you have any idea about why external displays aren’t working? It’s a simple issue, miniDP or VGA don’t respond at all, anyhow. (I know VGA isn’t easy to get working but point is, external display support is dead). Could USB mapping help fixing this issue?

The primary issue keeping me from using this EFI is the touch screen, but these are the two major issues which led me to switch to windows 11 after one day of use. Really need to switch back.

**Comment by MultimediaLucario** [comment]
Variants with a touchscreen do require addtional patching. Please read the Post-Install Tweaks section of my guide for more information.

**Reply by thomas-catt** [reply]
Okay so I have updates. I installed Monterey once again, but this time I pulled the SMBIOS from **valnoxy/t440p-oc**, which fixed the dead miniDP. I'm dailying this EFI now.

However, I attempted display patching to fix touch, but doing so resulted in my display turning off while booting. I'm sure I did something wrong because I had confusions while following your Display Patching guide. I couldn't find help on Google either.

- On step 3 of the guide, it says to _Go to Detailed Data tab and change H. Sync Width: value to 100._ But here I have two **blocks** here of some sort, do you have any idea about what may be causing this or if this is normal?
![image](https://user-images.githubusercontent.com/75628498/226163941-c8587de0-c819-492a-b3db-0affdffb4c4e.png)

- Step 6, _Copy the 128 bytes code and paste it._ Do I copy and paste the entire row of the codes?
![image](https://user-images.githubusercontent.com/75628498/226164069-ddb2c6fd-d6db-4b7a-a2f0-118261beb674.png)
 
  Or just one cluster from them?
  ![image](https://user-images.githubusercontent.com/75628498/226164189-a7f07f4d-595b-4881-a20c-c830707ac7a5.png)

Hope you're able to help me with these. Also, looking forward to updates from your Ventura EFI. Were you able to get WiFi working in it?

**Reply by MultimediaLucario** [reply]
I wasn't able to get Wi-Fi working in Ventura without using this app known as HeliPort.

**Comment by MultimediaLucario** [comment]
I am currently doing research and according to the OpenCore Guide, it seems as if VoodooI2CHID might be the answer to getting the touchscreen to work in macOS.

**Comment by MultimediaLucario** [comment]
Or you could try enabling VoodooPS2's VoodooInput kext.

**Comment by MultimediaLucario** [comment]
Diagnosing this is a bit more difficult because my T440s is out of commission.
